RZTouchID/RZTouchID.m

```diff
@@ -121,44 +144,33 @@ - (void)addPassword:(NSString *)password withIdentifier:(NSString *)identifier c
 
 - (void)retrievePasswordWithIdentifier:(NSString *)identifier withPrompt:(NSString *)prompt completion:(RZTouchIDCompletion)completion
 {
-    if ( [[self class] touchIDAvailable] ) {
-        CFErrorRef error = NULL;
-        SecAccessControlRef accessObject;
-        
-        accessObject = SecAccessControlCreateWithFlags(kCFAllocatorDefault, kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly, kSecAccessControlUserPresence, &error);
-        
-        //Check if password exists
-        NSDictionary *query = @{
-                                (__bridge id)kSecClass: (__bridge id)kSecClassGenericPassword,
-                                (__bridge id)kSecAttrService: [self serviceNameForIdentifier:identifier],
-                                (__bridge id)kSecReturnData: @YES,
-                                (__bridge id)kSecUseOperationPrompt: prompt,
-                                (__bridge id)kSecAttrAccessControl: (__bridge id)accessObject
-                                };
-        
-        
-        dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^(void){
-            CFTypeRef passwordData = NULL;
-            
-            OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)(query), &passwordData);
-            
-            NSString *password = [[NSString alloc] initWithData:(__bridge NSData *)passwordData encoding:NSUTF8StringEncoding];
-            NSError *error = [self errorForOsStatus:status];
-            
-            CFRelease(accessObject);
-            if ( passwordData != NULL ) {
-                CFRelease(passwordData);
-            }
+    //New context each time to avoid cached responses
+    LAContext *laContext = [LAContext new];
+    laContext.localizedFallbackTitle = self.localizedFallbackTitle;
```

Comment:
Can't the LAContext creation go inside the `if ( self.touchIDMode ==  RZTouchIDModeLocalAuthentication )` ?